drivers/net/ppp.c

```diff
@@ -176,7 +188,11 @@ static void ppp_change_state(struct ppp_driver_context *ctx,
 static int ppp_send_flush(struct ppp_driver_context *ppp, int off)
 {
 	if (!IS_ENABLED(CONFIG_NET_TEST)) {
-		uart_pipe_send(ppp->send_buf, off);
+		u8_t *buf = ppp->send_buf;
+
+		do {
+			uart_poll_out(ppp->dev, *buf++);
+		} while (--off);
```

Comment:
The final call to `ppp_send_flush` in `ppp_send` can pass `off=0`. If that happens then `off` goes negative, the loop keeps transmitting from outside the buffer.

```suggestion
		while (off--) {
			uart_poll_out(ppp->dev, *buf++);
		};
```

Reply:
Thanks, fixed.